File: scala-pro/src/main/scala/systems/vostok/algo/bigsort/pure/SortRandomIntegers.scala
package systems.vostok.algo.bigsort.pure

import scala.collection.mutable.ArrayBuffer

object SortRandomIntegers extends App {
  val size: Int = 45000000
  val array = ArrayBuffer[Int]()

  (0 to size).foreach(it =>
    array += (Math.random() * it).toInt
  )

  array.foreach(println)
}

File: groovy-pro/src/main/java/systems/vostok/dump/pure/SortRandomIntegers.java
package systems.vostok.dump.pure;

import java.util.stream.IntStream;

public class SortRandomIntegers {
    public static void main(String[] args) {
        /*IntStream.range(0, 45000000)
                .parallel()
                .map(i -> (int) (Math.random() * i))
                .sorted()
                .forEachOrdered(System.out::println);*/


        IntStream.range(0, 45000000)
                .map(i -> (int) (Math.random() * i))
                .sorted()
                .forEach(System.out::println);
    }
}
